### PR TITLE
Update type annotations in `/pyvista/trame`

### PR DESCRIPTION
**Pull request by user27182**
### Overview

With the `from __future__ import annotations` import added in  #5712, it should be possible to upgrade all type annotations to make use of syntax from [PEP 585](https://peps.python.org/pep-0585/) and [PEP 604](https://peps.python.org/pep-0604/), for example.

This PR updates the config for ~`pyupgrade`~ `ruff` to implement the changes.

EDIT: The upgrades for most of the repository have been merged in #6235. The scope of this PR is now reduced to files in `/pyvista/trame`

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (tkoyama010) [inline .pre-commit-config.yaml:86]: `pyupgrade` also affects syntax other than typing. Therefore, I was concerned about possible syntax errors in python<3.10. However, I decide it is not too late to take care of it after something happen.

Comment (user27182) [reply .pre-commit-config.yaml:86]: We could drop `pyupgrade` altogether and use ruff instead, see with https://docs.astral.sh/ruff/rules/#pyupgrade-up. This way it may be easier to control different options rather than having it as a kind of all-or-nothing with `pyupgrade`. This would be similar to dropping isort from #5712.

Comment (user27182) [reply .pre-commit-config.yaml]: This has been reverted to `--py38-plus`. With the `--keep-runtime-typing` option, `ruff` will treat fixes such as [UP006](https://docs.astral.sh/ruff/rules/non-pep585-annotation/) as "unsafe" and won't automatically apply the fixes. This is a good thing, since in some (very few/relatively uncommon) cases we need the runtime behavior (e.g. for some aliases we need to use `Tuple`, and not `tuple`). With the `--py38-plus` oftion ruff will simply complain about using `Tuple` but won't autofix this.

**Comment by tkoyama010** [comment]
Thank you. This makes the type annotation easier to read.

**Review by tkoyama010 — DISMISSED** [review]

File: pyvista/core/dataset.py
Comment (user27182) [inline pyvista/core/dataset.py:45]:
```suggestion
if TYPE_CHECKING:  # pragma: no cover
```

**Comment by tkoyama010** [comment]
We will merge this after merge #6159.

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Review by tkoyama010 — APPROVED** [review]

**Review by tkoyama010 — DISMISSED** [review]

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: pyvista/core/grid.py
Comment (tkoyama010) [inline pyvista/core/grid.py:5]:
```suggestion
```

File: pyvista/examples/_dataset_loader.py
Comment (tkoyama010) [inline pyvista/examples/_dataset_loader.py:36]:
```suggestion
```

File: pyvista/plotting/charts.py
Comment (tkoyama010) [inline pyvista/plotting/charts.py:5]:
```suggestion
```

**Review by tkoyama010 — DISMISSED** [review]

File: pyproject.toml
Comment (user27182) [inline pyproject.toml:97]: This can be reverted, I don't think it was intentional and I may have confused it with the target version for `ruff`.
```suggestion
target-version = ['py39']
```

**Review by banesullivan — APPROVED** [review]

**Pull request merged**
